desktop/package/windows/Bisq.iss

```diff
@@ -64,6 +65,96 @@ begin
   Result := False;
 end;
 
+procedure DirectoryCopy(SourcePath, DestPath: string);
+var
+  FindRec: TFindRec;
+  SourceFilePath: string;
+  DestFilePath: string;
+begin
+  if FindFirst(SourcePath + '\*', FindRec) then
+  begin
+    try
+      repeat
+        if (FindRec.Name <> '.') and (FindRec.Name <> '..') then
+        begin
+          SourceFilePath := SourcePath + '\' + FindRec.Name;
+          DestFilePath := DestPath + '\' + FindRec.Name;
+          if FindRec.Attributes and FILE_ATTRIBUTE_DIRECTORY = 0 then
+          begin
+            if FileCopy(SourceFilePath, DestFilePath, False) then
+            begin
+              Log(Format('Copied %s to %s', [SourceFilePath, DestFilePath]));
+            end
+              else
+            begin
+              Log(Format('Failed to copy %s to %s', [SourceFilePath, DestFilePath]));
+            end;
+          end
+            else
+          begin
+            if DirExists(DestFilePath) or CreateDir(DestFilePath) then
+            begin
+              Log(Format('Created %s', [DestFilePath]));
+              DirectoryCopy(SourceFilePath, DestFilePath);
+            end
+              else
+            begin
+              Log(Format('Failed to create %s', [DestFilePath]));
+            end;
+          end;
+        end;
+      until not FindNext(FindRec);
+    finally
+      FindClose(FindRec);
+    end;
+  end
+    else
+  begin
+    Log(Format('Failed to list %s', [SourcePath]));
+  end;
+end;
+
+//Delete old app directory to prevent issues during update
+procedure DeleteOldAppDataDirectory;
+var
+  entry: String;
+begin
+  entry := ExpandConstant('{localappdata}') + '\Bisq\';
+  if DirExists(entry) then begin
+    DelTree(entry, true, true, true);
+  end
+end;
+
+procedure DeleteTorFiles;
+var
+  mainnetDir: String;
+  torDir: String;
+  hiddenServiceDir: String;
+  hiddenServiceBackupDir : String;
+begin
+  mainnetDir := ExpandConstant('{userappdata}') + '\Bisq\btc_mainnet';
+  torDir := mainnetDir + '\tor\*';
+  hiddenServiceDir := mainnetDir + '\tor\hiddenservice';
+  hiddenServiceBackupDir := mainnetDir + '\hiddenservice_backup';
+  if DirExists(hiddenServiceDir) then begin
+   if DirExists(hiddenServiceBackupDir) then begin
+     DelTree(hiddenServiceBackupDir, true, true, true);
+   end
+   CreateDir(hiddenServiceBackupDir);
+   DirectoryCopy(hiddenServiceDir, hiddenServiceBackupDir);
+   DelTree(torDir, false, true, true);
+   CreateDir(hiddenServiceDir);
+   DirectoryCopy(hiddenServiceBackupDir, hiddenServiceDir);
```

Comment:
Should we remove hiddenServiceBackupDir once we are done with it?

Reply:
I left it there by intent, just in case there is an issue and someone has to recover their hiddenservices manually.